oneandone/src/main/java/org/apache/jclouds/oneandone/rest/features/ServerApi.java

```diff
@@ -0,0 +1,567 @@
+import org.jclouds.rest.annotations.Fallback;
+import org.jclouds.rest.annotations.MapBinder;
+import org.jclouds.rest.annotations.PayloadParam;
+import org.jclouds.rest.annotations.RequestFilters;
+import org.jclouds.rest.annotations.ResponseParser;
+import org.jclouds.rest.binders.BindToJsonPayload;
+import org.jclouds.util.Strings2;
+
+@Path("servers")
+@RequestFilters(AuthenticateRequest.class)
+public interface ServerApi extends Closeable {
+
+    @Named("servers:list")
+    @GET
+    @ResponseParser(ServerApi.ServerListParser.class)
+    @Fallback(Fallbacks.EmptyListOnNotFoundOr404.class)
+    List<Server> getList();
+
+    @Named("servers:list")
+    @GET
+    @ResponseParser(ServerApi.ServerListParser.class)
+    @Fallback(Fallbacks.EmptyListOnNotFoundOr404.class)
+    List<Server> getList(GenericQueryOptions options);
+
+    @Named("servers:Flavours:list")
+    @GET
+    @Path("/fixed_instance_sizes")
+    @ResponseParser(ServerApi.HardwareFlavoursParser.class)
+    @Fallback(Fallbacks.EmptyListOnNotFoundOr404.class)
+    List<HardwareFlavour> listHardwareFlavours();
+
+    @Named("servers:Flavours:get")
+    @GET
+    @Path("fixed_instance_sizes/{id}")
+    @ResponseParser(ServerApi.HardwareFlavourParser.class)
+    @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+    HardwareFlavour getHardwareFlavour(@PathParam("id") String flavourId);
+
+    @Named("servers:get")
+    @GET
+    @Path("/{serverId}")
+    @ResponseParser(ServerApi.ServerParser.class)
+    @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+    Server get(@PathParam("serverId") String serverId);
+
+    @Named("servers:status:get")
+    @GET
+    @Path("/{serverId}/status")
+    @ResponseParser(ServerApi.StatusParser.class)
+    @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+    Status getStatus(@PathParam("serverId") String serverId);
+
+    @Named("servers:create")
+    @POST
+    @ResponseParser(ServerApi.ServerParser.class)
+    Server create(@BinderParam(BindToJsonPayload.class) CreateServer server);
+
+    @Named("servers:create")
+    @POST
+    @ResponseParser(ServerApi.ServerParser.class)
+    Server createFixedInstanceServer(@BinderParam(BindToJsonPayload.class) CreateFixedInstanceServer server);
+
+    @Named("server:update")
+    @PUT
+    @Path("/{serverId}")
+    @ResponseParser(ServerApi.ServerParser.class)
+    @Produces("application/json")
+    Server update(@PathParam("serverId") String serverId, @BinderParam(BindToJsonPayload.class) Server.UpdateServer server);
+
+    @Named("server:Status:update")
+    @PUT
+    @Path("/{serverId}/status/action")
+    @ResponseParser(ServerApi.ServerParser.class)
+    @Produces("application/json")
+    Server updateStatus(@PathParam("serverId") String serverId, @BinderParam(BindToJsonPayload.class) Server.UpdateStatus server);
+
+    @Named("server:delete")
+    @DELETE
+    @Path("/{serverId}")
+    @ResponseParser(ServerApi.ServerParser.class)
+    @MapBinder(BindToJsonPayload.class)
```

Comment:
This did work for the DELETE method it didn't generate he content-type header for me
My code example 
```
@Named("servers:privatenetwork:delete")
    @DELETE
    @Path("/{serverId}/private_networks/{privateNetworkId}")
    @Fallback(Fallbacks.NullOnNotFoundOr404.class)
    @Produces("application/json")
    Server deletePrivateNetwork(@PathParam("serverId") String serverId, @PathParam("privateNetworkId") String privateNetworkId);
```

this is what the header looks like :

`User-Agent: jclouds-okhttp/2.0.0-SNAPSHOT java/1.8.0_73
Accept: application/json
X-TOKEN: 4f34bcc24bf7bbf6af2fac5e35e600d8
Content-Length: 0
Host: cloudpanel-api.1and1.com
Connection: Keep-Alive
Accept-Encoding: gzip
`